src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -312,116 +233,24 @@ namespace OpenMS
-            delta_int = intensity - intensity_prev;
-            height_50 = intensity - 0.5*peak_height;
-            peakShapeMetrics.start_time_at_50 = retention_time - delta_int*delta_rt/height_50;
-          }
-        } 
-        else if (retention_time > peak_apex_rt)
-        {
-          // end_time_at_5
-          if (intensity <= 0.05*peak_height && 
-            intensity_prev > 0.05*peak_height)
-          {
-            delta_rt = retention_time - retention_time_prev;
-            delta_int = intensity_prev - intensity;
-            height_5 = 0.05*peak_height - intensity;
-            peakShapeMetrics.end_time_at_5 = retention_time - delta_int*delta_rt/height_5;
-          }
-          // start_time_at_10
-          if (intensity <= 0.1*peak_height && 
-            intensity_prev > 0.1*peak_height)
-          {
-            delta_rt = retention_time - retention_time_prev;
-            delta_int = intensity_prev - intensity;
-            height_10 = 0.1*peak_height - intensity;
-            peakShapeMetrics.end_time_at_10 = retention_time - delta_int*delta_rt/height_10;
-          }
-          // end_time_at_50
-          if (intensity <= 0.5*peak_height && 
-          intensity_prev > 0.5*peak_height)
-          {
-            delta_rt = retention_time - retention_time_prev;
-            delta_int = intensity_prev - intensity;
-            height_50 = 0.5*peak_height - intensity;
-            peakShapeMetrics.end_time_at_50 = retention_time - delta_int*delta_rt/height_50;
-          }
-        } 
-
-        // points across the peak
-        peakShapeMetrics.points_across_baseline ++;
-        if (intensity >= 0.5*peak_height)
-        {
-          peakShapeMetrics.points_across_half_height ++;
-        }
-      }
-    }
-
-    // peak widths
-    peakShapeMetrics.width_at_5 = peakShapeMetrics.end_time_at_5 - peakShapeMetrics.start_time_at_5;
-    peakShapeMetrics.width_at_10 = peakShapeMetrics.end_time_at_10 - peakShapeMetrics.start_time_at_10;
-    peakShapeMetrics.width_at_50 = peakShapeMetrics.end_time_at_50 - peakShapeMetrics.start_time_at_50;
-    peakShapeMetrics.total_width = best_right - best_left;
-    peakShapeMetrics.slope_of_baseline = end_intensity - start_intensity;
-    peakShapeMetrics.baseline_delta_2_height = peakShapeMetrics.slope_of_baseline / peak_height;
-
-    // other
-    peakShapeMetrics.tailing_factor = peakShapeMetrics.width_at_5 / std::min(peak_apex_rt - peakShapeMetrics.start_time_at_5, peakShapeMetrics.end_time_at_5 - peak_apex_rt);
-    peakShapeMetrics.asymmetry_factor = std::min(peak_apex_rt - peakShapeMetrics.start_time_at_10, peakShapeMetrics.end_time_at_10 - peak_apex_rt) / std::max(peak_apex_rt - peakShapeMetrics.start_time_at_10, peakShapeMetrics.end_time_at_10 - peak_apex_rt);
+    PeakIntegrator pi;
+    PeakIntegrator::PeakShapeMetrics psm = pi.calculatePeakShapeMetrics(chromatogram, best_left, best_right, peak_height, peak_apex_rt);
```

Comment:
I would think that all background/integration calls should look something like this.  (i.e., a call to a method in PeakIntegrator, and then an extraction of all variables needed by MRMTransitionGroupPicker)

Reply:
This is what I've done, except for `calculateBgEstimationAverage_()` which I did not touch. Should I also modify that one?

Reply:
The average function uses a slightly different method to calculate the background.  It would probably be best to add that method to PeakIntegrator and then call using the PeakIntegrator interface.